fastcrypto-zkp/src/unit_tests/api_tests.rs

```diff
@@ -1,171 +1,113 @@
-        &ALPHA_G1_BETA_G2_BYTES,
-        &GAMMA_G2_NEG_PC_BYTES,
-        &DELTA_G2_NEG_PC,
-        &PROOF_INPUTS_BYTES,
-        &PROOF_POINTS_BYTES
+        &bytes[0],
+        &bytes[1],
+        &bytes[2],
+        &bytes[3],
+        &proof_inputs_bytes,
+        &proof_points_bytes
     )
     .is_ok());
 
     // Length of verifying key is incorrect.
-    let mut modified_bytes = VK_GAMMA_ABC_G1_VECTOR.to_vec();
+    let mut modified_bytes = bytes[0].clone();
     modified_bytes.pop();
     assert!(verify_groth16_in_bytes(
         &modified_bytes,
-        &ALPHA_G1_BETA_G2_BYTES,
-        &GAMMA_G2_NEG_PC_BYTES,
-        &DELTA_G2_NEG_PC,
-        &PROOF_INPUTS_BYTES,
-        &PROOF_POINTS_BYTES
+        &bytes[1],
+        &bytes[2],
+        &bytes[3],
+        &proof_inputs_bytes,
+        &proof_points_bytes
     )
     .is_err());
 
     // Length of public inputs is incorrect.
-    let mut modified_proof_points_bytes = PROOF_INPUTS_BYTES.to_vec();
-    modified_proof_points_bytes.pop();
+    let mut modified_proof_inputs_bytes = proof_inputs_bytes.clone();
+    modified_proof_inputs_bytes.pop();
     assert!(verify_groth16_in_bytes(
-        &VK_GAMMA_ABC_G1_VECTOR,
-        &ALPHA_G1_BETA_G2_BYTES,
-        &GAMMA_G2_NEG_PC_BYTES,
-        &DELTA_G2_NEG_PC,
-        &modified_proof_points_bytes,
-        &PROOF_POINTS_BYTES
+        &modified_bytes,
+        &bytes[1],
+        &bytes[2],
+        &bytes[3],
+        &modified_proof_inputs_bytes,
+        &proof_points_bytes
     )
     .is_err());
 
     // length of proof is incorrect
-    let mut modified_proof_points_bytes = PROOF_POINTS_BYTES.to_vec();
+    let mut modified_proof_points_bytes = proof_points_bytes.to_vec();
     modified_proof_points_bytes.pop();
     assert!(verify_groth16_in_bytes(
-        &VK_GAMMA_ABC_G1_VECTOR,
-        &ALPHA_G1_BETA_G2_BYTES,
-        &GAMMA_G2_NEG_PC_BYTES,
-        &DELTA_G2_NEG_PC,
-        &PROOF_INPUTS_BYTES,
+        &modified_bytes,
+        &bytes[1],
+        &bytes[2],
+        &bytes[3],
+        &proof_inputs_bytes,
         &modified_proof_points_bytes
     )
     .is_err());
 }
 
 #[test]
 fn test_prepare_pvk_bytes() {
+    const PUBLIC_SIZE: usize = 128;
+    let rng = &mut thread_rng();
+    let c = DummyCircuit::<Fr> {
+        a: Some(<Fr>::rand(rng)),
+        b: Some(<Fr>::rand(rng)),
+        num_variables: PUBLIC_SIZE,
+        num_constraints: 10,
+    };
+
+    let (_, vk) = Groth16::<Bls12_381>::circuit_specific_setup(c, rng).unwrap();
+
+    let mut vk_bytes = vec![];
+    vk.serialize_compressed(&mut vk_bytes).unwrap();
+
     // Success case.
-    assert!(prepare_pvk_bytes(&VK_BYTES).is_ok());
+    assert!(prepare_pvk_bytes(vk_bytes.as_slice()).is_ok());
 
     // Length of verifying key is incorrect.
-    let mut modified_bytes = VK_BYTES.to_vec();
+    let mut modified_bytes = vk_bytes.clone();
     modified_bytes.pop();
     assert!(prepare_pvk_bytes(&modified_bytes).is_err());
 }
-
```

Comment:
The existing test vector gave an error (not in subgroup), so I recreated a random test from the arkworks api instead.

Reply:
Need to understand better, what was the issue that it was failing?

Reply:
Something may have changed in the way arkworks does serialization or deserialization, because the old test vector didn't work anymore. So I rewrote the test such that it generates a test vector on the fly instead, and this worked.